File: 0001_0100/0034/0034.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, k = map(int, f.readline().split())
    res = "NO"
    if n > k:
        s = f.readline()
        l = {int(s[:k]):0}
        for i in range(1, n - k + 1):
            d = int(s[i : k + i])
            if d in l:
                res = "YES"
                break
            l[d] = 0
    q.write(res)

File: 0001_0100/0034/0034.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n, k int
	var s string
	r := "NO"
	fmt.Fscan(f, &n, &k, &s)
	m := make(map[string]int)
	for i := 0; i <= n-k; i++ {
		m[s[i:i+k]] = 1
		if len(m) == i {
			r = "YES"
			break
		}
	}
	fmt.Fprint(q, r)
}
